nano/node/bootstrap_ascending/service.cpp

```diff
@@ -150,7 +151,7 @@ void nano::bootstrap_ascending::service::inspect (store::transaction const & tx,
 		break;
 		case nano::block_status::gap_source:
 		{
-			const auto account = block.previous ().is_zero () ? block.account_field ().value () : ledger.account (tx, block.previous ()).value ();
+			const auto account = block.previous ().is_zero () ? block.account_field ().value () : view.account (block.previous ()).value ();
```

Comment:
Ledger view should be passed instead of database transaction, otherwise data races are possible. Assuming bounded backlog is implemented, with current code it's possible that unconfirmed blocks will be dropped in between calls to inspect (). Another aspect is that creating a new view for each call is very suboptimal.

Reply:
To clarity, this is not an issue right now thanks to guarantees provided by database transactions, but will be once unconfirmed set is no longer stored on disk.

Reply:
To address the allocations concern, these changes were redesigned.